plugins/widget/plugin.js

```diff
@@ -1066,17 +1064,11 @@
 		 * @returns {Boolean} Whether the style is applied to this widget.
 		 */
 		checkStyleActive: function( style ) {
-			var classes = getStyleClasses( style ),
-				cl;
-
-			if ( !classes )
-				return false;
+			style = CKEDITOR.tools.copy( style );
+			style.element = this.element.getName();
+			style._.definition.ignoreReadonly = true;
```

Comment:
Here is already an example the modifying copy (`style._.definition.ignoreReadonly`), will also modify source object. See below:

![image](https://user-images.githubusercontent.com/1061942/36488699-6a088718-1724-11e8-91ff-e58823e79ded.png)